lhotse/supervision.py

```diff
@@ -15,6 +15,8 @@ class SupervisionSegment:
     language: Optional[str] = None
     speaker: Optional[str] = None
     gender: Optional[str] = None
+    # exmaple: "data/graph/train_si284/num.1.ark:9"
+    graph_archive: Optional[str] = None
```

Comment:
I think we should keep the graph loading logic more self-contained. How about: 

```
pychain_graph: Optional[PyChainGraph] 
```

and:

```
@dataclass
class PyChainGraph:
    archive_path: Pathlike
    archive_offset: int

    def from_dict(data: dict) -> 'PyChainGraph':
        ...  # look up other from_dict implementations across the codebase
    
    def load():
        import simplefst
        from pychain import ChainGraph
        # loads and returns a chain graph; 
        # local import ensures that only the users who actually need it will need to have pychain/simplefst installed
```

you'll need to update `SupervisionSegment.from_dict` to make sure that dict is converted to this class during deserialization

Reply:
We might also need fields like `state_offset: int` and `num_states_to_load: int` (somewhat equivalent to `begin: Seconds` and `duration: Seconds`)

Reply:
Sounds good to me!

Reply:
Given our discussion, the `state_offset` and `num_states_to_load` shouldn't be need after all.